blocks/collections-content-source-block/sources/content-api-collections.js

```
@@ -9,6 +9,21 @@ const params = {
 	size: "text",
 };
 
+const handleError = (error) => {
+	if (error?.response) {
+		// eslint-disable-next-line no-console
+		console.error(
+			`The response from the server was an error with the status code ${error?.response?.status}.`
+		);
+	} else if (error?.request) {
+		// eslint-disable-next-line no-console
+		console.error("The request to the server failed with no response.");
+	} else {
+		// eslint-disable-next-line no-console
+		console.error("An error occured creating the request.");
+	}
+};
```

Wondering if we can't just return the strings from these instead of console logging... testing.  The error state of the promise is returned, but we could return the text so pagebuilder renders a friendly error?

---

I thought about that, but the useContent calls are expecting an object or undefined to be returned from the content sources. So it could cause some errors in our blocks due to the unexpected data type.

---

Looks like we can just throw the error text instead of logging it (bottom code block, "Fixed Example"): https://docs.arcxp.com/alc?sys_kb_id=5bae879347aedd10d0cba8b5536d4316&id=kb_article_view#Incorrect-Usage

---

okay, that makes sense, I'll make that update in a minute.

---

Update pushed.